simpeg_drivers/params.py

```diff
@@ -103,6 +100,15 @@ class CoreOptions(BaseData):
     out_group: SimPEGGroup | UIJsonGroup | None = None
     generate_sweep: bool = False
 
+    @field_validator("z_from_topo", mode="before")
+    @classmethod
+    def deprecated(cls, v, info):
+        warnings.warn(
+            f"Field '{info.field_name}' is deprecated. Since version 0.3.0, "
+            f"any data location adjustments must be done in pre-processing."
+        )
```

Comment:
```suggestion
        logger.warning(
            f"Field %s is deprecated. Since version 0.3.0, "
            f"any data location adjustments must be done in pre-processing.", info.field_name
        )
```